misk/src/test/kotlin/misk/flags/FlagPropertiesTest.kt

```diff
@@ -0,0 +1,105 @@
+package misk.flags
+
+import com.google.inject.name.Named
+import misk.flags.memory.InMemoryDynamicFlagStore
+import misk.flags.memory.InMemoryDynamicFlagStoreModule
+import misk.inject.KAbstractModule
+import misk.moshi.MoshiModule
+import misk.testing.ActionTest
+import misk.testing.ActionTestModule
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.data.Offset
+import org.junit.jupiter.api.Test
+import javax.inject.Inject
+
+@ActionTest(startService = false)
+class FlagPropertiesTest {
+
+    data class JsonData(val message: String)
+
+    class MyFlags internal constructor(context: Flags.Context) : Flags("my-flags", context) {
+        val string by StringFlag("This is the string flag", default = "defValue")
+        val int by IntFlag("this is the int flag", default = 101)
+        val boolean by BooleanFlag("this is the boolean flag", default = true)
+        val double by DoubleFlag("this is the double flag", default = 100.45)
+        val json by JsonFlag<JsonData>(description = "this is the json flag")
+    }
+
+    @ActionTestModule
+    val testModule = object : KAbstractModule() {
+        override fun configure() {
+            install(MoshiModule())
+            install(InMemoryDynamicFlagStoreModule())
+            install(object : DynamicFlagModule() {
+                override fun configureFlags() {
+                    bindFlags<MyFlags>()
+                    bindFlags<MyFlags>("unique-prefix")
+                }
+            })
+        }
+    }
+
+    @Inject
+    private
+    lateinit
+    @Named("unique-prefix")
+    var prefixedFlags: MyFlags
```

Comment:
Eventually I’d like a way to programmatically get this without using the Injector. For example, for archiver we don’t know the prefix statically.